src/encoder/encoderopus.h

```diff
@@ -0,0 +1,61 @@
+// encoderopus.h
+// Create on August 15th 2017 by Palakis
+
+#ifndef ENCODER_ENCODEROPUS_H
+#define ENCODER_ENCODEROPUS_H
+
+#include <QMap>
+#include <QString>
+#include <QVector>
+
+#include <ogg/ogg.h>
+#include <opus/opus.h>
+
+#include "encoder/encoder.h"
+#include "encoder/encodercallback.h"
+#include "util/fifo.h"
+#include "util/sample.h"
+#include "util/samplebuffer.h"
+
+class EncoderOpus: public Encoder {
+  public:
+    static const int MASTER_SAMPLERATE;
+    static const char* INVALID_SAMPLERATE_MESSAGE;
+
+    EncoderOpus(EncoderCallback* pCallback = nullptr);
+    virtual ~EncoderOpus();
+
+    int initEncoder(int samplerate, QString errorMessage) override;
+    void encodeBuffer(const CSAMPLE *samples, const int size) override;
+    void updateMetaData(const QString& artist, const QString& title, const QString& album) override;
+    void flush() override;
+    void setEncoderSettings(const EncoderSettings& settings) override;
+
+  private:
+    static QString opusErrorString(int error);
```

Comment:
Those two private static functions don't need access to OpusEncoder and can be hidden in the anonymous namespace